tests/server.rs

```diff
@@ -1841,117 +1839,117 @@ async fn h2_connect() {
-                .body(())
-                .unwrap();
-            let (response, send_stream) = h2.send_request(request, false).unwrap();
-            streams.push((i, response, send_stream));
-        }
-
-        let futures = streams
-            .into_iter()
-            .map(|(i, response, mut send_stream)| async move {
-                if i % 4 == 0 {
-                    return;
-                }
-
-                let response = response.await.unwrap();
-                assert_eq!(response.status(), StatusCode::OK);
-
-                if i % 4 == 1 {
-                    return;
-                }
-
-                let mut body = response.into_body();
-                let bytes = body.data().await.unwrap().unwrap();
-                assert_eq!(&bytes[..], b"Bread?");
-                let _ = body.flow_control().release_capacity(bytes.len());
-
-                if i % 4 == 2 {
-                    return;
-                }
-
-                send_stream.send_data("Baguette!".into(), true).unwrap();
-
-                assert!(body.data().await.unwrap().unwrap().is_empty());
-            })
-            .collect::<FuturesUnordered<_>>();
-
-        futures.for_each(future::ready).await;
-    });
-
-    let svc = service_fn(move |req: Request<Body>| {
-        let authority = req.uri().authority().unwrap().to_string();
-        let on_upgrade = hyper::upgrade::on(req);
-
-        tokio::spawn(async move {
-            let upgrade_res = on_upgrade.await;
-            if authority == "localhost_0" {
-                assert!(upgrade_res.expect_err("upgrade cancelled").is_canceled());
-                return;
-            }
-            let mut upgraded = upgrade_res.expect("upgrade successful");
-
-            upgraded.write_all(b"Bread?").await.unwrap();
-
-            let mut vec = vec![];
-            let read_res = upgraded.read_to_end(&mut vec).await;
-
-            if authority == "localhost_1" || authority == "localhost_2" {
-                let err = read_res.expect_err("read failed");
-                assert_eq!(err.kind(), io::ErrorKind::Other);
-                assert_eq!(
-                    err.get_ref()
-                        .unwrap()
-                        .downcast_ref::<h2::Error>()
-                        .unwrap()
-                        .reason(),
-                    Some(h2::Reason::CANCEL),
-                );
-                return;
-            }
-
-            read_res.unwrap();
-            assert_eq!(s(&vec), "Baguette!");
-
-            upgraded.shutdown().await.unwrap();
-        });
-
-        future::ok::<_, hyper::Error>(
-            Response::builder()
-                .status(200)
-                .body(hyper::Body::empty())
-                .unwrap(),
-        )
-    });
-
-    let (socket, _) = listener.accept().await.unwrap();
-    Http::new()
-        .http2_only(true)
-        .serve_connection(socket, svc)
-        .with_upgrades()
-        .await
-        .unwrap();
-}
+// #[tokio::test]
```

Comment:
I believe this is testing some important behavior for HTTP/2, and that it's using `FuturesUnordered` is fine. Could keep this test mostly as-is.